iroh-net-report/src/reportgen/probes.rs

```diff
@@ -263,46 +273,52 @@ impl ProbePlan {
                     })
                     .expect("adding QuicIpv6 probe to a QuicAddrIpv6 probe set");
             }
-            plan.add(stun_ipv4_probes);
-            plan.add(stun_ipv6_probes);
-            plan.add(quic_ipv4_probes);
-            plan.add(quic_ipv6_probes);
+            plan.add_if_enabled(stun_ipv4_probes);
+            plan.add_if_enabled(stun_ipv6_probes);
+            plan.add_if_enabled(quic_ipv4_probes);
+            plan.add_if_enabled(quic_ipv6_probes);
 
             // The HTTP and ICMP probes only start after the STUN probes have had a chance.
             let mut https_probes = ProbeSet::new(ProbeProto::Https);
-            let mut icmp_probes_ipv4 = ProbeSet::new(ProbeProto::IcmpV4);
-            let mut icmp_probes_ipv6 = ProbeSet::new(ProbeProto::IcmpV6);
+            let mut icmp_v4_probes = ProbeSet::new(ProbeProto::IcmpV4);
+            let mut icmp_v6_probes = ProbeSet::new(ProbeProto::IcmpV6);
+
             for attempt in 0..3 {
-                let start = *max_stun_delay.get_or_insert_with(|| plan.max_delay())
-                    + DEFAULT_INITIAL_RETRANSMIT;
+                let mut start = *max_high_prio_delay.get_or_insert_with(|| plan.max_delay());
+                // if there are high priority probes, ensure there is a buffer between
+                // the highest probe delay and the next probes we create
+                // if there are no high priority probes, we don't need a buffer
+                if plan.has_priority_probes() {
```

Comment:
This is indeed much more readable, thanks!

Can this condition now be `if !start.is_zero()` which would avoid this extra function and categorizing probes?

Reply:
I thought about that, but was trying to future-proof:

basically, if we do some refactor that means that it's possible for only 1 probe to be attempted, that probe will be at delay == 0, which would make `plan.max_delay` == 0

Reply:
Oh, right.  Didn't think of that case, yes your approach here makes the most sense then.